test/data/collection.cpp

```diff
@@ -326,31 +326,35 @@ BOOST_AUTO_TEST_CASE(collection__is_sorted__unsorted__false)
 
 BOOST_AUTO_TEST_CASE(collection__distinct__empty__empty)
 {
-    BOOST_REQUIRE(distinct(data_chunk{}).empty());
+    data_chunk data{};
+    BOOST_REQUIRE(distinct(data).empty());
```

Comment:
This changes the test from move to copy overload. Maybe the macro would be happy if the result was computed externally.

Reply:
Reverted test changes with Collection signatures altered to `Collection foo(Collection&&)` signature.